### ISSUE TITLE
no-mergeable-selectors incorrectly trying to merge two selectors in different mixins

### ISSUE THREAD

**Issue by bgriffith** [sasstools/sass-lint#339]
This rule incorrectly tries to merge two selectors within two different mixins.
SassLint: 1.3.1
```scss
@mixin foo {
    &:disabled {
        color: red;
    }
}

@mixin input-bar {
    &:disabled {
        cursor: blue;
    }
}
```

**Comment by DanPurdy** [comment]
Yeah, i need to look at mixins as part of #307 anyway so I'll combine the effort then

**Comment by bgriffith** [comment]
Ah yeah, I see they're related.

**Comment by brokentone** [comment]
Same issue here. Although a slightly different test case for me -- one inside a mixin.

```
    & a { 
      display: block;
    }   

    @include breakpoint($bp-760) {
      & a { 
        display: inline-block;
      }   
    }   
```

**Comment by DanPurdy** [comment]
Yeah, I'm hoping to look at this soon, it's going to be a little bit of a refactor it looks like.

**Comment by tonifisler** [comment]
Same issue here with media queries:

```scss
.spacing {
  margin: 16px 0;
  &.spacing-sm {margin: 8px 0;}
  @media only screen and (max-width: 767px) {
    margin: 8px 0;
    &.spacing-sm {margin: 5px 0;}
  }
}

.spacing-bottom {
  margin-bottom: 32px;
  &.spacing-sm {margin-bottom: 15px;}
  @media only screen and (max-width: 767px) {
    margin-bottom: 15px;
    &.spacing-sm {margin-bottom: 10px;} // sass-lint triggers here
    // and suggests merging with the rule in the media query of .spacing
  }
}
```

**Comment by DanPurdy** [comment]
Don't worry, this is in progress and should be resolved in our 1.6 update which we're hoping to get to you soon.

**Comment by DanPurdy** [comment]
@tonifisler @bgriffith @brokentone All of your examples are fixed in the rewrite of the mergeable selectors rule released with sass-lint 1.6.0!
